web/src/components/Visualization/components/DAG/SpanNode.tsx

```diff
@@ -1,27 +1,31 @@
 import {ClockCircleOutlined, SettingOutlined, ToolOutlined} from '@ant-design/icons';
-import {Space} from 'antd';
-import {useMemo, useState} from 'react';
+import {useMemo} from 'react';
 import {Handle, NodeProps, Position} from 'react-flow-renderer';
 
-import {SemanticGroupNamesToText} from 'constants/SemanticGroupNames.constants';
-import {SpanKindToText} from 'constants/Span.constants';
-import {useAppSelector} from 'redux/hooks';
-import SpanService from 'services/Span.service';
-import TestSpecsSelectors from 'selectors/TestSpecs.selectors';
-import {INodeDataSpan} from 'types/DAG.types';
 import AssertionResultChecks from 'components/AssertionResultChecks/AssertionResultChecks';
-import {selectOutputsBySpanId} from 'redux/testOutputs/selectors';
+import CurrentSpanSelector from 'components/CurrentSpanSelector';
 import TestOutputMark from 'components/TestOutputMark';
 import {useTestSpecForm} from 'components/TestSpecForm/TestSpecForm.provider';
-import CurrentSpanSelector from 'components/CurrentSpanSelector';
+import {SemanticGroupNamesToText} from 'constants/SemanticGroupNames.constants';
+import {SpanKindToText} from 'constants/Span.constants';
 import {useSpan} from 'providers/Span/Span.provider';
 import {useTestOutput} from 'providers/TestOutput/TestOutput.provider';
 import {useTestRun} from 'providers/TestRun/TestRun.provider';
+import {useAppDispatch, useAppSelector} from 'redux/hooks';
+import {selectAnalyzerResults} from 'redux/slices/Trace.slice';
+import {selectOutputsBySpanId} from 'redux/testOutputs/selectors';
+import TestSpecsSelectors from 'selectors/TestSpecs.selectors';
+import TraceSelectors from 'selectors/Trace.selectors';
+import SpanService from 'services/Span.service';
+import {INodeDataSpan} from 'types/DAG.types';
+import AnalyzerResults from './AnalyzerResults';
+import * as SA from './AnalyzerResults.styled';
 import * as S from './SpanNode.styled';
 
 interface IProps extends NodeProps<INodeDataSpan> {}
 
 const SpanNode = ({data, id, selected}: IProps) => {
+  const dispatch = useAppDispatch();
```

Comment:
We need to consider cleaning up nodes to not have to be connected to all of this state and such, maybe something to think latter on

Reply:
Yes, agree with that. I have been thinking about that for a while. We need to parse everything (span related) and inject it in the data object.